packages/surveys/src/components/wrappers/StackedCardsContainer.tsx

```diff
@@ -0,0 +1,147 @@
+  const cardBorderColor = survey.styling?.cardBorderColor?.light || styling.cardBorderColor?.light;
+  const cardRefs = useRef<(HTMLDivElement | null)[]>([]);
+  const resizeObserver = useRef<ResizeObserver | null>(null);
+  const [cardHeight, setCardHeight] = useState("auto");
+
+  const cardIndexes = useMemo(() => {
+    let cardIndexTemp = survey.questions.map((_, index) => index);
+    if (survey.welcomeCard.enabled) {
+      cardIndexTemp.unshift(-1);
+    }
+    if (survey.thankYouCard.enabled) {
+      cardIndexTemp.push(survey.questions.length);
+    }
+    return cardIndexTemp;
+  }, [survey]);
+
+  const questionIdx = useMemo(() => {
+    if (currentQuestionId === "start") return survey.welcomeCard.enabled ? -1 : 0;
+    if (currentQuestionId === "end") return survey.thankYouCard.enabled ? survey.questions.length : 0;
+    return survey.questions.findIndex((question) => question.id === currentQuestionId);
+  }, [currentQuestionId, survey.welcomeCard.enabled, survey.thankYouCard.enabled]);
+
+  const borderStyles = useMemo(() => {
+    const baseStyle = {
+      border: "1px solid",
+      borderRadius: "var(--fb-border-radius)",
+    };
+    // Determine borderColor based on the survey type and availability of highlightBorderColor
+    const borderColor =
+      survey.type === "link" || !highlightBorderColor ? cardBorderColor : highlightBorderColor;
+    return {
+      ...baseStyle,
+      borderColor: borderColor,
+    };
+  }, [survey.type, cardBorderColor, highlightBorderColor]);
+
+  const calculateCardTransform = useMemo(() => {
+    return (offset: number) => {
+      switch (cardArrangement) {
+        case "casual":
+          return offset < 0 ? `translateX(33%)` : `translateX(0) rotate(-${(hovered ? 3.5 : 3) * offset}deg)`;
+        case "straight":
+          return offset < 0 ? `translateY(25%)` : `translateY(-${(hovered ? 12 : 10) * offset}px)`;
+        default:
+          return offset < 0 ? `translateX(0)` : `translateX(0)`;
+      }
+    };
+  }, [cardArrangement, hovered]);
+
+  const straightCardArrangementStyles = (offset: number) => {
+    if (cardArrangement === "straight") {
+      // styles to set the descending width of stacked question cards when card arrangement is set to straight
+      return {
+        width: `${100 - 5 * offset}%`,
+        margin: "auto",
+      };
+    }
+  };
+
+  // UseEffect to handle the resize of current question card and set cardHeight accordingly
+  useEffect(() => {
+    const currentElement = cardRefs.current[questionIdx];
+    if (currentElement) {
+      if (resizeObserver.current) resizeObserver.current.disconnect();
+      resizeObserver.current = new ResizeObserver((entries) => {
+        for (const entry of entries) setCardHeight(entry.contentRect.height + "px");
+      });
+      resizeObserver.current.observe(currentElement);
+    }
+    return () => resizeObserver.current?.disconnect();
+  }, [questionIdx, cardRefs.current]);
```

Comment:
please check the linter warning and either extend the deps array (ideal case) or add an ignore statement